internal/pkg/externalplugins/blunderbuss/blunderbuss.go

```diff
@@ -108,13 +111,45 @@ func HandlePullRequestEvent(gc githubClient, pe *github.PullRequestEvent,
 	opts := cfg.BlunderbussFor(repo.Owner.Login, repo.Name)
 
 	// Wait a few seconds to allow other automation plugin to apply labels.
-	gracePeriod := DefaultGracePeriodDuration * time.Second
+	gracePeriod := time.Duration(opts.GracePeriodDuration) * time.Second
+	sleep(gracePeriod)
+
+	return handle(
+		gc,
+		opts,
+		repo,
+		pr,
+		log,
+		ol,
+	)
+}
+
+// handlePullRequestLabeledEvent handles the sig label labeled event.
+func handlePullRequestLabeledEvent(gc githubClient, pe *github.PullRequestEvent,
+	cfg *externalplugins.Configuration, ol ownersclient.OwnersLoader, log *logrus.Entry) error {
+	pr := &pe.PullRequest
+	repo := &pe.Repo
+	opts := cfg.BlunderbussFor(repo.Owner.Login, repo.Name)
 
-	if opts.GracePeriodDuration != 0 {
-		gracePeriod = time.Duration(opts.GracePeriodDuration) * time.Second
+	// Only handle the event of adding the SIG label of open PR.
+	if !(pe.Action == github.PullRequestActionLabeled && pe.PullRequest.State == "open" &&
```

Comment:
I think no need check action here and maybe we should move the state check into `HandlePullRequestEvent`.